src/routes/internal/client/profiles.rs

```diff
@@ -0,0 +1,1292 @@
+                    .execute(&mut *transaction)
+                    .await?;
+                }
+            }
+
+            // Edit the metadata fields
+            if edit_data.loader_version.is_some() || edit_data.game_version.is_some() {
+                let mut metadata = data.metadata.clone();
+
+                match &mut metadata {
+                    client_profile_item::ClientProfileMetadata::Minecraft {
+                        loader_version,
+                        game_version_id,
+                        game_version,
+                    } => {
+                        if let Some(new_loader_version) = edit_data.loader_version {
+                            *loader_version = new_loader_version;
+                        }
+
+                        if let Some(new_game_version) = edit_data.game_version {
+                            let new_game_id =
+                                database::models::legacy_loader_fields::MinecraftGameVersion::list(
+                                    None, None, &**pool, &redis,
+                                )
+                                .await?
+                                .into_iter()
+                                .find(|x| x.version == new_game_version)
+                                .ok_or_else(|| {
+                                    ApiError::InvalidInput(
+                                        "Invalid Client game version".to_string(),
+                                    )
+                                })?
+                                .id;
+
+                            *game_version_id = new_game_id;
+                            *game_version = new_game_version;
+                        }
+                    }
+                    client_profile_item::ClientProfileMetadata::Unknown => {
+                        return Err(ApiError::InvalidInput(
+                            "Cannot edit metadata of unknown profile".to_string(),
+                        ));
+                    }
+                }
+
+                sqlx::query!(
+                    "UPDATE shared_profiles SET metadata = $1 WHERE id = $2",
+                    serde_json::to_value(metadata)?,
+                    data.id.0
+                )
+                .execute(&mut *transaction)
+                .await?;
+            }
+
+            transaction.commit().await?;
+            client_profile_item::ClientProfile::clear_cache(data.id, &redis).await?;
+            return Ok(HttpResponse::NoContent().finish());
+        } else {
+            return Err(ApiError::CustomAuthentication(
+                "You are not the owner of this profile".to_string(),
+            ));
+        }
+    }
+    Err(ApiError::NotFound)
+}
+
+// Delete a client profile
+pub async fn profile_delete(
+    req: HttpRequest,
+    info: web::Path<(String,)>,
+    pool: web::Data<PgPool>,
+    redis: web::Data<RedisPool>,
+    session_queue: web::Data<AuthQueue>,
+) -> Result<HttpResponse, ApiError> {
```

Comment:
This should delete override files from the CDN as long as they are unused by other shared packs